### PR TITLE
Feat/nuclear trajectories

### PR DESCRIPTION
**Pull request by pworschischek-aggmag**
closes https://github.com/AGGM-AG/pypsa-at-planning/issues/158

## Changes proposed in this Pull Request
INtroduces PEMMDB trajectories for `nuclear` powerplants. `nuclear` carrier is a conventional technology and not by default extendable. This PR add extendable assets for all countries with nuclear powerplants and applied trajectory bands frpm the `TRAJECTORIES.xlsx` PEMMDB data file to them.

nuclear trajectories are given per TYNDP scenario in the input file. Setting `mods.PEMMDB_trajectories.tyndp_scenario='All'` will use the smallest and largest values from all scenarios as `p_nom_min` and `p_nom_max` boundaries. It is possible to select specific scenarios though. 

## Checklist

**Required:**
- [x] Changes are tested locally and behave as expected.
- [x] Code and workflow changes are documented.
- [x] A brief description of the changes has been added to `Changelog.md`.
- [x] All Sourcery Bot review suggestions have been implemented or discarded with an explanation.
- [x] All github actions succeed.

## Summary by Sourcery

Introduce PEMMDB nuclear capacity trajectories and ensure they are consistently applied to extendable nuclear assets, with supporting scenario handling, validations, and documentation updates.

New Features:
- Add support for extendable nuclear links so PEMMDB TYNDP capacity trajectories can bound nuclear capacity expansion.
- Introduce collapsing of per-scenario nuclear trajectories into a single trajectory per location and planning year based on configured TYNDP scenario.
- Add pytest configuration to make scripts and hyphenated subdirectories importable without per-test path hacks.

Bug Fixes:
- Fix trajectory brownfield deduction for at_port=1 carriers to use the correct port units and avoid efficiency-related errors.

Enhancements:
- Guard that each sited asset is unique per component, location, carrier, and build year to protect vintage-based logic.
- Fail fast if any extendable asset configured for PEMMDB trajectories is left without finite capacity bounds.
- Clarify and reorganize README usage, quickstart, and feature section descriptions for PyPSA-AT.
- Tighten nuclear trajectory handling to sum brownfield electrically, clamp negative bounds to zero, and require matching trajectory data for all nuclear locations.

Build:
- Wire TYNDP scenario selection into the build_tyndp_trajectories Snakemake rule so nuclear scenario collapsing can be configured.

Documentation:
- Document quickstart, usage, prerequisites, and external documentation links more clearly in the README.
- Update changelog entry to note that nuclear is now covered by enforced Open-TYNDP capacity trajectories.

Tests:
- Add unit tests for nuclear PEMMDB trajectory application, nuclear extendable link registration, and brownfield handling under multiple vintages and edge cases.
- Add regression tests for port-consistent trajectory brownfield deduction and for ensuring one vintage asset per (component, location, carrier, build_year).
- Add tests for collapsing nuclear TYNDP trajectories across scenarios and for trajectory ceilings including nuclear carriers.

### CODE REVIEW

File: scripts/open-tyndp/build_tyndp_trajectories.py [hunk 1]
```diff
@@ -20,6 +20,61 @@
...
+def collapse_nuclear_scenarios(df: pd.DataFrame, tyndp_scenario: str) -> pd.DataFrame:
+    """
+    Reduce per-scenario nuclear trajectories to a single row per (bus, pyear).
+
+    Unlike the RES/storage carriers (which only carry the scenario-independent
+    ``"All"`` label), TYNDP provides nuclear trajectories per scenario
+    (``DE``/``GA``/``NT``).  Downstream readers do not filter by scenario, so
+    nuclear must be collapsed to one row per location and horizon:
+
+    - ``tyndp_scenario == "All"``: take the smallest ``p_nom_min`` and the
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply scripts/open-tyndp/build_tyndp_trajectories.py:33]: non-issue. We follow a fail-fast philosphy

File: scripts/open-tyndp/build_tyndp_trajectories.py [hunk 2]
```diff
@@ -20,6 +20,61 @@
...
+            "pypsa_eur_carrier",
+            "open_tyndp_type",
+        ]
+        nuclear = (
+            nuclear.groupby(group_keys, as_index=False)
+            .agg(p_nom_min=("p_nom_min", "min"), p_nom_max=("p_nom_max", "max"))
+            .assign(scenario="All")
+        )
+    else:
+        nuclear = nuclear[nuclear["scenario"] == tyndp_scenario]
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply scripts/open-tyndp/build_tyndp_trajectories.py:77]: checkd tyndp-scenario valid inputs upfron

File: README.md [hunk 3]
```diff
@@ -32,116 +32,101 @@ and [PyPSA-DE documentation](https://ariadneprojekt.de/modell-dokumentation-pyps
 
 ---
 
+## 🚀 Quickstart 
+Running PyPSA-AT is very simple. Just clone repository
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -32,116 +32,101 @@ and [PyPSA-DE documentation](https://ariadneprojekt.de/modell-dokumentation-pyps
...
+git clone https://github.com/AGGM-AG/pypsa-at.git && cd pypsa-at
+```
+And start the workflow (low time resolution and NUTS3 spatial resolution for Austria)
+```sh
+pixi run workflow
+```
+
+## ⌨️ Usage
+Please note that PyPSA-AT is only supported on **Linux** platforms. Installations on Windows or macOS require
+modifications currently not supported.
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -32,116 +32,101 @@ and [PyPSA-DE documentation](https://ariadneprojekt.de/modell-dokumentation-pyps
...
+
+### Prerequisites
+[Git](https://git-scm.com/install) and [pixi](https://pixi.prefix.dev/latest/#installation) must be installed.
+
+### Intstallation
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply README.md]: done

File: README.md [hunk 6]
```diff
@@ -32,116 +32,101 @@ and [PyPSA-DE documentation](https://ariadneprojekt.de/modell-dokumentation-pyps
...
 [documentation](https://pypsa-at.readthedocs.io/en/latest/).
 
-### 🧭 Scenarios
-- 🔨 Under active development
+### 🧭 Scenarios (🔨 Under active development)
 
-### 🧮 Carbon Accounting
-- 🔨 Under active development
+### 🧮 Carbon Accounting (🔨 Under active development)
+- National CO2 budgets for exogen decarbonization paths
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply README.md]: done

File: docs-at/explanations/brownfield/open-tyndp.md [hunk 7]
```diff
@@ -60,6 +61,31 @@ so `existing_brownfield` reflects only assets that remain economically active.
 For the 2025 base year `p_nom_min` is set to zero for all non-nuclear carriers. The 2025 investment
 optimum should not be pre-determined by a floor constraint.
 
+### Nuclear: a conventional carrier
```
Comment (nicolezechneraggm) [inline docs-at/explanations/brownfield/open-tyndp.md:64]: use of 'vintage' as description of Links, maybe specify further?

File: docs-at/explanations/brownfield/open-tyndp.md [hunk 8]
```diff
@@ -60,6 +61,31 @@ so `existing_brownfield` reflects only assets that remain economically active.
...
+`build_year` set to the current horizon. `apply_trajectories` then writes the `p_nom_min` / `p_nom_max`
+bounds onto it at port 1 (electricity output), exactly as for the other Link carriers. The vintages are
+left untouched and carry the brownfield as a fixed floor.
+
+!!! warning "Myopic years only"
+    Nuclear stays non-extendable in the 2025 base year, mirroring PyPSA-Eur's treatment of conventional
+    technologies. The extendable Link is synthesized and bounded **only in myopic horizons** (2030
+    onward); no extendable nuclear is created and no nuclear trajectory is applied for the base year.
+
+Because an *unbounded* extendable nuclear Link would let the solver build unlimited nuclear,
```
Comment (nicolezechneraggm) [inline docs-at/explanations/brownfield/open-tyndp.md:83]: personal preference: 'build unlimited nuclear' sounds unfinished; I'd use 'nuclear potential' or 'nuclear capacity' or something similar if applicable.

File: mods/network/potentials.py [hunk 9]
```diff
@@ -248,6 +228,106 @@ def apply_trajectories(
...
+    n: pypsa.Network,
+    traj: pd.DataFrame,
+    skip_countries: tuple[str],
+    pyear: int,
+) -> None:
+    """
+    Add one extendable nuclear Link per location so :func:`apply_trajectories` can bound it.
+
+    Nuclear is a conventional carrier represented by one or more non-extendable
+    *vintage* Links per location (``bus0="EU uranium"``, ``bus1`` = electricity
```
Comment (nicolezechneraggm) [inline mods/network/potentials.py:241]: again, use of vintage as description, but that might just be a me-problem

File: docs-at/explanations/brownfield/open-tyndp.md (hunk 7 above)
Comment (pworschischek-aggmag) [reply docs-at/explanations/brownfield/open-tyndp.md:64]: explained further at one point. Bypassed with more known nomenclature elsewhere

File: docs-at/explanations/brownfield/open-tyndp.md (hunk 8 above)
Comment (pworschischek-aggmag) [reply docs-at/explanations/brownfield/open-tyndp.md:83]: changed to "build unlimited nuclear capacity, [...]"

File: mods/network/potentials.py (hunk 9 above)
Comment (pworschischek-aggmag) [reply mods/network/potentials.py:241]: Didn't touch it in the docstrings because the term refers to one specific asset while the term "brownfield" describes all vintage assets together. I think the difference matters and we will nedd the term vintage in the future to name single brownfield assets. 

vintages in this case refers to pre-optimization brownfield assets, e.g. ``FR nuclear-2010`` or similar. They carry capacity from before the optimization into the planning_horizon year. They have a `build_year` that effectively retires them at `-{year}` suffix plus `build_year`.

**Review by nicolezechneraggm — APPROVED** [review]
looks good to me
